src/SWP/ContentBundle/Twig/Extension/ContentExtension.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\ContentBundle\Twig\Extension;
+
+use SWP\ContentBundle\Document\Article;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\Routing\Router;
+
+class ContentExtension extends \Twig_Extension
+{
+    /**
+     * Entity Manager.
+     *
+     * @var EntityManager
+     */
+    protected $em;
+
+    /**
+     * Router.
+     *
+     * @var Router
+     */
+    protected $router;
+
+    public function __construct(EntityManager $em, Router $router)
+    {
+        $this->em = $em;
+        $this->router = $router;
+    }
+
+    public function getFunctions()
+    {
+        return array(
+            'generateUrlFor*' => new \Twig_SimpleFunction('generateUrlFor*', array($this, 'generateUrlFor')),
+        );
+    }
+
+    public function getName()
+    {
+        return 'swp_content';
+    }
+
+    public function generateUrlFor($name, $object)
+    {
+        if (is_object($object) && method_exists($object, 'getValues')) {
+            if ($name == 'Article' && $object->getValues() instanceof Article) {
+                $pageArticle = $this->em->getRepository('SWP\ContentBundle\Model\PageContent')
+                    ->getForContentPath($object->getValues()->getId())
```

Comment:
`getForContentPath` is not clear, it doesnt tell me what you want to get, query or some value for the contentPath in the form of a query object  etc. that should be `getContentPathQuery()` or if we assume that repository will return only objects of type `Query,` we could simply name it `getContentPath()`